woocommerce/woocommerce-ios
# Coupons: Add category selector

Part of #6490


### Description

This PR adds the category selector to the Add/Edit Coupon screen and Coupon Restrictions screen.
Changes:
- Wrap `ProductCategoryListViewController` in a SwiftUI view named `ProductCategoryList`.
- Update `ProductCategoryListViewModel` to accept selected IDs and fetch categories immediately upon initializing to avoid ghost animations.
- Add a new SwiftUI view `ProductCategorySelector` that contains `ProductCategoryList`.
- Present `ProductCategorySelector` from Add/Edit Coupon screen.
- Present `ProductCategorySelector` from Coupon Restrictions screen.

### Testing instructions

- Navigate to Menu > Coupons > Select a coupon > Select the ellipsis button > Edit Coupon
- If the coupon is restricted to any category, notice that the category button is displayed as "Edit Product Categories <number of items>". Otherwise, it displays "Select Product Categories".
- Select the category button, notice that a new screen is presented with a list of categories to select from. Any pre-selected categories should be displayed with a check icon.
- Try select/unselect any item to see that the Done button displays the correct number of selected items. If none is selected, it should simply show "Done".
- Tap the Done button, notice that the category button is correct on the Edit Coupon screen.
- Select Usage Restrictions - notice that you can see the Exclude Categories screen when tapping the Exclude Categories button.

### Screenshots

https://user-images.githubusercontent.com/5533851/165919620-46c6d124-0962-4783-8b0f-2a6850badbf2.mp4



---
